routers/web/repo/release.go

```diff
@@ -68,7 +70,11 @@ func Releases(ctx *context.Context) {
 
 // TagsList render tags list page
 func TagsList(ctx *context.Context) {
-	releasesOrTags(ctx, true)
+	if ctx.Repo.CanAccess(perm.AccessModeRead, unit.TypeReleases) {
+		releasesOrTags(ctx, true)
+	} else if ctx.Repo.CanAccess(perm.AccessModeRead, unit.TypeCode) {
```

Comment:
Maybe the permission check here is unnecessary? as we have `reqRepoCodeReader` in
https://github.com/go-gitea/gitea/pull/23465/files#diff-3f5521b3af1ddd518b1958e0799d7ab00254d39af04eb96dc00d59d2644eb30fR1179

Reply:
Fixed in https://github.com/go-gitea/gitea/pull/23465/commits/81895bf2f69a0900db48f4216f19fceb883e0b66